### PR TITLE
mod: reduce bark/shout spam

### PR DESCRIPTION
**Pull request by MrMendo**

### CODE REVIEW

File: src/Module.Client/GUI/CrpgMissionGauntletMainAgentCheerControllerView.cs [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+    private MissionMainAgentCheerBarkControllerVM? _dataSource;
+
+    private float _cooldownTimeRemaining;
+
+    private int _barkUsesLeft;
+
+    private float _barkUsesResetCooldownTime;
+
+    private bool _isSelectingFromInput;
+
```
Comment (Muparadzi) [inline src/Module.Client/GUI/CrpgMissionGauntletMainAgentCheerControllerView.cs:50]: Why so much whitespace!

File: src/Module.Client/GUI/CrpgMissionGauntletMainAgentCheerControllerView.cs [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+        {
+            _holdHandled = value;
+            MissionScreen?.SetRadialMenuActiveState(value);
+        }
+    }
+
+    public CrpgMissionGauntletMainAgentCheerControllerView()
+    {
+        _gauntletLayer = new GauntletLayer(2);
+        _dataSource = new MissionMainAgentCheerBarkControllerVM(OnCheerSelect, OnBarkSelect, Agent.TauntCheerActions, SkinVoiceManager.VoiceType.MpBarks);
```
Comment (Muparadzi) [inline src/Module.Client/GUI/CrpgMissionGauntletMainAgentCheerControllerView.cs:82]: Why are these not in `OnMissionScreenInitialize()`?

**Review by Muparadzi — CHANGES_REQUESTED** [review]

File: src/Module.Client/GUI/CrpgMissionGauntletMainAgentCheerControllerView.cs (hunk 1 above)
Comment (MrMendo) [reply src/Module.Client/GUI/CrpgMissionGauntletMainAgentCheerControllerView.cs]: Ah, forgot about that one.
Was just copy pasted decompiled code and never gave another tought about that.

File: src/Module.Client/GUI/CrpgMissionGauntletMainAgentCheerControllerView.cs (hunk 2 above)
Comment (MrMendo) [reply src/Module.Client/GUI/CrpgMissionGauntletMainAgentCheerControllerView.cs]: Good catch.
First i wanted to solve the issue without making those nullable but decided later not to.

**Comment by MrMendo** [comment]
As a help.
Most code is native decompiled code.

Those fields are used in the custom implementation.
private float _cooldownTimeRemaining;
private int _barkUsesLeft;
private float _barkUsesResetCooldownTime;

File: src/Module.Client/GUI/CrpgMissionGauntletMainAgentCheerControllerView.cs [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+                else if (MissionScreen.SceneLayer.Input.IsHotKeyReleased("CheerBarkSelectSecondCategory"))
+                {
+                    if (_barkUsesLeft > 0)
+                    {
+                        _dataSource.SelectItem(1);
+                    }
+                    else
+                    {
+                        _barkCooldownTimeInfoText.SetTextVariable("SECONDS", _barkUsesResetCooldownTime.ToString("0.0"));
+                        InformationManager.DisplayMessage(new InformationMessage(_barkCooldownTimeInfoText.ToString()));
```
Comment (namidaka) [inline src/Module.Client/GUI/CrpgMissionGauntletMainAgentCheerControllerView.cs:234]: Can you confirm that first category is cheers and second one is barks?

**Review by namidaka — CHANGES_REQUESTED** [review]

File: src/Module.Client/GUI/CrpgMissionGauntletMainAgentCheerControllerView.cs [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+
+    public override void OnMissionScreenFinalize()
+    {
+        base.OnMissionScreenFinalize();
+        Mission.OnMainAgentChanged -= OnMainAgentChanged;
+        MissionScreen.RemoveLayer(_gauntletLayer);
+        _gauntletLayer = null;
+        _dataSource?.OnFinalize();
+        _dataSource = null;
+    }
```
[the remark this answers is not in the document]
Comment (MrMendo) [reply src/Module.Client/GUI/CrpgMissionGauntletMainAgentCheerControllerView.cs]: you are right
readded it just in case

**Pull request merged**